### PR TITLE
Remove distinction between auth- and resource-server

### PR DESCRIPTION
**Pull request by tkrille**
The core servers are merged now, so the connector has to catch up with that. Instead of 3 separate methods for setting endpoints, there will be only 1 left, namely `OsiamConnector#setEndpoint`, to set the endpoint of OSIAM. Also removed the ability to configure the maximum connections per route, because there will be only 1 route from now on.

See osiam/osiam#43, osiam/osiam#44

### CODE REVIEW

**Comment by tkrille** [comment]
Changes in docs are still missing.

File: src/main/java/org/osiam/client/OsiamConnector.java [hunk 1]
```diff
@@ -62,29 +60,25 @@
...
 
-    private static final PoolingHttpClientConnectionManager CONNECTION_MANAGER;
-    private static final javax.ws.rs.client.Client client;
     public static final ObjectMapper objectMapper = new ObjectMapper();
 
-    static {
-        CONNECTION_MANAGER = new PoolingHttpClientConnectionManager();
-        CONNECTION_MANAGER.setMaxTotal(DEFAULT_MAX_CONNECTIONS);
-        CONNECTION_MANAGER.setDefaultMaxPerRoute(DEFAULT_MAX_CONNECTIONS_PER_ROUTE);
+    private static final PoolingHttpClientConnectionManager CONNECTION_MANAGER =
```
Comment (dacrome) [inline src/main/java/org/osiam/client/OsiamConnector.java]: Please rename to `connectionManager` because this isn't a traditional constant.

Comment (tkrille) [inline src/main/java/org/osiam/client/OsiamConnector.java]: It actually is, it's only used as a reference and never together with the dot operator.

Comment (tkrille) [inline src/main/java/org/osiam/client/OsiamConnector.java]: Oh, I see where it is not :)

Comment (tkrille) [inline src/main/java/org/osiam/client/OsiamConnector.java]: done

**Comment by tkrille** [comment]
Docs are updated. This can be merged now.

**Comment by dacrome** [comment]
You forgot some OSIAM server mentions:
- https://github.com/tkrille/osiam-connector4java/blob/support-merge/README.md
- [create-osiam-connector.md](https://github.com/tkrille/osiam-connector4java/blob/support-merge/docs/create-osiam-connector.md): Please rename `SERVER_ENDPOINT_ADDRESS` to `OSIAM_ENDPOINT_ADDRESS`
- ...

Just search for "server" or "OSIAM server" there are also occurrences in javadoc, e.g. in [AuthService](https://github.com/tkrille/osiam-connector4java/blob/support-merge/src/main/java/org/osiam/client/AuthService.java#L590)

**Comment by tkrille** [comment]
Done

**Comment by tkrille** [comment]
Made additional 2 changes.

**Comment by dacrome** [comment]
Awesome! Thank you!

**Pull request merged**